packages/ckeditor5-core/src/context.ts

```diff
@@ -160,11 +162,36 @@ export default class Context {
 
 		const languageConfig = this.config.get( 'language' ) || {};
 
-		this.locale = new Locale( {
-			uiLanguage: typeof languageConfig === 'string' ? languageConfig : languageConfig.ui,
-			contentLanguage: this.config.get( 'language.content' ),
-			translations
-		} );
+		if ( !translations && global.window.CKEDITOR_TRANSLATIONS ) {
+			/**
+			 * Function _translate from translation-service.ts gets translations from
+			 * global.window.CKEDITOR_TRANSLATIONS when translations injected into Locale are empty.
+			 * Value of global.window.CKEDITOR_TRANSLATIONS is provided by CKEditorTranslationsPlugin from
+			 * ckeditor5-dev-translations package (for example in manual tests) when --language flag is used.
+			 * Function _translate is called often and has no access to the editing editor config, so here is a better place to
+			 * check if translations will be taken from dev-translations, and if yes – update config.language.ui value.
+			 */
```

Comment:
imho, this comment could be clearer. Two things:

1. It sounds like `global.window.CKEDITOR_TRANSLATIONS` is _always_ set by `CKEditorTranslationsPlugin`, but that’s not true. It can also come from translation files loaded from a CDN, so we shouldn’t assume it always comes from the plugin.
2. Without reading the issue ([#7297](https://github.com/cksource/ckeditor5-commercial/issues/7297)), it’s hard to tell what the problem is and why this function is needed. The comment should explain it in a more straightforward way.